File: backend/src/main/java/com/example/IOT_SmartStick/service/GeofenceService.java
package com.example.IOT_SmartStick.service;

import com.example.IOT_SmartStick.dto.request.GeofenceRequestDTO;
import com.example.IOT_SmartStick.dto.response.GeofenceResponseDTO;

import java.util.List;

public interface GeofenceService {
    GeofenceResponseDTO createGeofence(Long deviceId, GeofenceRequestDTO request, String token);
    GeofenceResponseDTO updateGeofence(Long geofenceId, GeofenceRequestDTO request, String token);
    void deleteGeofence(Long geofenceId, String token);
    GeofenceResponseDTO getGeofenceById(Long geofenceId, String token);
    List<GeofenceResponseDTO> getGeofencesByDevice(Long deviceId, String token);
    GeofenceResponseDTO toggleGeofenceStatus(Long geofenceId, String token);
}

File: frontend/src/service/geofenceService.js
// service/geofenceService.js
import axios from "./axiosCustomize";

/**
 * Tạo geofence mới
 */
const createGeofence = (deviceId, geofenceData) => {
  const token = localStorage.getItem("accessToken");
  return axios.post(`/api/geofence/device/${deviceId}`, geofenceData, {
    headers: { Authorization: `Bearer ${token}` },
  });
};

/**
 * Lấy danh sách geofence của device
 */
const getGeofencesByDevice = (deviceId) => {
  const token = localStorage.getItem("accessToken");
  return axios.get(`/api/geofence/device/${deviceId}`, {
    headers: { Authorization: `Bearer ${token}` },
  });
};

/**
 * Cập nhật geofence
 */
const updateGeofence = (geofenceId, geofenceData) => {
  const token = localStorage.getItem("accessToken");
  return axios.put(`/api/geofence/${geofenceId}`, geofenceData, {
    headers: { Authorization: `Bearer ${token}` },
  });
};

/**
 * Bật/tắt geofence
 */
const toggleGeofence = (geofenceId) => {
  const token = localStorage.getItem("accessToken");
  return axios.patch(`/api/geofence/${geofenceId}/toggle`, {}, {
    headers: { Authorization: `Bearer ${token}` },
  });
};

/**
 * Xóa geofence
 */
const deleteGeofence = (geofenceId) => {
  const token = localStorage.getItem("accessToken");
  return axios.delete(`/api/geofence/${geofenceId}`, {
    headers: { Authorization: `Bearer ${token}` },
  });
};

export {
  createGeofence,
  getGeofencesByDevice,
  updateGeofence,
  toggleGeofence,
  deleteGeofence,
};